compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -335,11 +349,11 @@ void RecordMinMax::profileRawData(const std::string &mode, const std::string &in
 
   std::cout << "Recording finished. Number of recorded data: " << num_records << std::endl;
 
-  update_quantparam(_observer.get(), mode, min_percentile, max_percentile);
+  update_quantparam(_observers[0].get(), mode, min_percentile, max_percentile);
 }
 
-void RecordMinMax::profileData(const std::string &mode, const std::string &input_data_path,
-                               float min_percentile, float max_percentile)
+std::vector<std::vector<std::vector<char>>>
```

Comment:
Please split this PR by purpose
- revise to prepare to multiple instance
- use openmp for running in multiple instance